packages/core/src/dispatcher.ts

```diff
@@ -46,24 +47,19 @@ const createActionDispatcher = <TModels extends Models>(
  * actions.
  */
 const createDispatcher = <
-	TModels extends Models<TModels>,
-	TModel extends NamedModel<TModels>
+	TModels extends Models<TModels> = Record<string, any>,
```

Comment:
great Record<string, any>

Reply:
for typescript 4.0.0 this is the best approach

Reply:
`Record` was great except this: 

```ts
type RematchStoreWithEmptyObject = RematchStore<RootModel & {}>
type RematchStoreWithRecord = RematchStore<RootModel & Record<string, any>>

const store1: RematchStoreWithEmptyObject
const store2: RematchStoreWithRecord

// (property) dolphins: any
// Type of property 'dolphins' circularly references itself in mapped type 'ExtractRematchDispatchersFromModels<RootModel & Record<string, any>>'.ts(2615)
// Type of property 'questions' circularly references itself in mapped type 'ExtractRematchDispatchersFromModels<RootModel & Record<string, any>>'.ts(2615)
// Type of property 'sharks' circularly references itself in mapped type 'ExtractRematchDispatchersFromModels<RootModel & Record<string, any>>'.ts(2615)
store2.dispatch.dolphins.increment

// worked
store1.dispatch.dolphins.increment
```

So I want to change the default generic value for `TExtraModels`, change from this: 

```ts
export declare const init: <
	TModels extends Models<TModels> = Record<string, any>,
	TExtraModels extends Models<TModels> = Record<string, any>
>(
	initConfig?: InitConfig<TModels, TExtraModels> | undefined
) => RematchStore<TModels & TExtraModels>
```

to this

```ts
export declare const init: <
	TModels extends Models<TModels> = Record<string, any>,
	TExtraModels extends Models<TModels> = {}
>(
	initConfig?: InitConfig<TModels, TExtraModels> | undefined
) => RematchStore<TModels & TExtraModels>
```

If our users don't use plugins that contain models, they don't need to pass `RootModel` to init, `RootModel` can be inferred automatically. But if they use Rematch with these plugin models, they need to pass `ExtraModels` info cause it can't be inferred. (Plugin config array contains these model Infos, but I can't find a way to iterate it, collect and combine all of them) And once they pass the `TExtraModel`, they need also pass `RootModel`, otherwise it will be inferred by its opinion value, but not from `initConfig`.

So pass `RootModel` and `ExtraModels` if use plugins that contain models, and don't need to pay attention if don't' use plugins or use plugins that just using hooks like `immer`.